### DISCUSSION TITLE
How to use profiling with thread pool

### DISCUSSION THREAD

**Discussion by xlq20080808** [apache/skywalking#9387] [Q&A] [answered]
I came across issue on profiling the thread pool endpoint.
I used apm-jdk-threadpool-plugin-8.12.0-SNAPSHOT.jar, at this time, the thread name in the thread pool will be used as the endpoint name. I want to use this endpoint for profiling, I expect to focus on one of the tasks, but this thread may perform other tasks
<img width="1545" alt="image" src="https://user-images.githubusercontent.com/33999699/181152037-b3f9da9a-fcf3-4066-9978-c02eb3fe33b3.png">
<img width="1513" alt="image" src="https://user-images.githubusercontent.com/33999699/181152202-08730095-5747-4d21-ab78-a716630840d8.png">

I have the following two questions
1.My understanding is that endpoint is from entry span. Why is the thread pool span recognized as an endpoint?
2.Doing profiling for the thread pool, I expect to be able to analyze the methods in the thread, but I can't do it at present. Is there a better way to do profiling for thread pool？

hope to be answered ,thx

**Comment by wu-sheng** [comment, marked as answer]
Profiling is segment oriented. Once you start a new thread or pool executor, yes, it is that. 
The name is created by the plugin you used, we don't know the difference.
You should creat span of the new executor with a custom span name, rather than using default plugin.

**Reply by xlq20080808** [reply]
Thanks for your answer, but I still have a question, how can I create span for executor. Is it similar to using @TraceCrossThread on thread classes, but there is no way to add custom names.

**Reply by wu-sheng** [reply]
I don't think you could control that with a custom operation name in that plugin. Because the plugin created the span before your codes ran. 
If you insist this is meaningful profiling, then you need a private plugin to create a new segment in the new thread. You have to learn the kernel APIs for the plugin.

**Reply by xlq20080808** [reply]
Thanks for your answer, it was helpful for me.
Currently I solved my problem in a way that doesn't seem very elegant. I modified apm-jdk-threadpool-plugin and put parentEndpointName into operationName, which can help me analyze only the target method during profiling

**Reply by wu-sheng** [reply]
The private solution doesn't concern the community. Once it works for you, that is good.
We have to provide a widely adaptable solution from an open-source project perspective.
